fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsDeleteBrowsingDataTest.kt

```diff
@@ -0,0 +1,256 @@
+    fun deleteBrowsingHistoryAndSiteDataTest() {
+        val storageWritePage = getStorageTestAsset(mockWebServer, "storage_write.html").url
+        val storageCheckPage = getStorageTestAsset(mockWebServer, "storage_check.html").url
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(storageWritePage) {
+        }.openNavigationToolbar {
+        }.enterURLAndEnterToBrowser(storageCheckPage) {
+            verifyPageContent("Session storage has value")
+            verifyPageContent("Local storage has value")
+        }.openThreeDotMenu {
+        }.openSettings {
+        }.openSettingsSubMenuDeleteBrowsingData {
+            verifyBrowsingHistoryDetails("2")
+            selectOnlyBrowsingHistoryCheckBox()
+            clickDeleteBrowsingDataButton()
+            clickDialogCancelButton()
+            verifyBrowsingHistoryDetails(true)
+            clickDeleteBrowsingDataButton()
+            confirmDeletionAndAssertSnackbar()
+            verifyBrowsingHistoryDetails("0")
+            exitMenu()
+        }
+        navigationToolbar {
+        }.openThreeDotMenu {
+        }.openHistory {
+            verifyEmptyHistoryView()
+            mDevice.pressBack()
+        }
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(storageCheckPage) {
+            verifyPageContent("Session storage empty")
+            verifyPageContent("Local storage empty")
+        }
+    }
+
+    @SmokeTest
+    @Test
+    fun deleteCookiesTest() {
+        val genericPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+        val cookiesTestPage = getStorageTestAsset(mockWebServer, "storage_write.html").url
+
+        // Browsing a generic page to allow GV to load on a fresh run
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(genericPage.url) {
+        }.openNavigationToolbar {
+        }.enterURLAndEnterToBrowser(cookiesTestPage) {
+            verifyPageContent("No cookies set")
+            clickSetCookiesButton()
+            verifyPageContent("user=android")
+        }.openThreeDotMenu {
+        }.openSettings {
+        }.openSettingsSubMenuDeleteBrowsingData {
+            selectOnlyCookiesCheckBox()
+            clickDeleteBrowsingDataButton()
+            confirmDeletionAndAssertSnackbar()
+            exitMenu()
+        }
+        browserScreen {
+        }.openThreeDotMenu {
+        }.refreshPage {
+            verifyPageContent("No cookies set")
+        }
+    }
+
+    @SmokeTest
+    @Test
+    fun deleteCachedFilesTest() {
+        val pocketTopArticles = getStringResource(R.string.pocket_pinned_top_articles)
+
+        homeScreen {
+            verifyExistingTopSitesTabs(pocketTopArticles)
+        }.openTopSiteTabWithTitle(pocketTopArticles) {
+            waitForPageToLoad()
+        }.openTabDrawer {
+        }.openNewTab {
+        }.submitQuery("about:cache") {
+            // disabling wifi to prevent downloads in the background
+            setNetworkEnabled(enabled = false)
```

Comment:
Does this work on Firebase Test Lab?

Reply:
yes, it's not a new test, I just moved it.